src/api/aaq/requests/inbounds.py

```
@@ -1,12 +1,33 @@
-from ast import literal_eval
+from collections.abc import Iterator
 
 from attrs import define
 from httpx import Client
-from pandas import DataFrame, concat
+from pandas import DataFrame
 
 from .. import get_paginated
 
 
+def build_faqranks(model_scoring_dict: dict) -> Iterator[dict]:
+    """Extracts important information from the AAQ model scoring dict.
+
+    Each inbound message contains a dict that contains the model scores
+    for every other faq not returned to the user. This function will
+    traverse this dict, look for nested dicts, and return only the
+    important information we care about for faqranks.
+
+    """
+    for k, v in model_scoring_dict.items():
+        if isinstance(v, dict):
```

This seems like a place where we might want to be more explicit about the structure of the data and fail if we get something unexpected rather than potentially returning the wrong thing. This is very much a data decision, though.

---

Yeah, was thinking the same thing. Let me commit a suggestion.